hibernate-core/src/test/java/org/hibernate/orm/test/associations/NotFoundTest.java

```diff
@@ -125,7 +125,7 @@ public void queryTestFk(EntityManagerFactoryScope scope) {
 					assertThat( nullResults ).isEmpty();
 
 					final List<String> nonNullResults = entityManager
-							.createQuery( "select p.name from Person p where fk( p.city ) is not null", String.class )
+							.createQuery( "select p.name from Person p left join p.city c where fk( c ) is not null", String.class )
```

Comment:
Is this change really necessary? I'd like to keep the original i.e. ensure using `fk()` on a path for not-found association has "left join semantics".

Reply:
The test runs 2 queries that are exactly the same, so the original check is still there. I only changed the second query to have an example of the new feature allowing use of `fk()` on an explicit join alias.

Reply:
It's not exactly the same (`is null` vs. `is not null`), but I guess it's ok since the end of this method verifies no joins are created.